Task 1

RamanReut opened the pull request:
Realization of first task

RamanReut committed:
Add redux store

RamanReut committed:
Complete basic layout

RamanReut committed:
Add Navbar expand button; Add breadcrumbs

RamanReut committed:
Add search field and avatar

RamanReut committed:
Add navbar; Some changes on header for better
responsive

RamanReut committed:
Add unit tests

dHihich commented on src/features/layout/Header.tsx:
```diff
@@ -0,0 +1,109 @@
+            </Grid>
+        </AppBar>
+    );
+}
```
no newline at the end
the same for majority of files. pls add eslint rule to check this

dHihich commented on src/features/layout/Header.test.tsx:
```diff
@@ -0,0 +1,77 @@
+
+function dummySearchTextChange (event: ChangeEvent<HTMLInputElement>): void {}
+
+test('header-searchInput', () => {
```
test names should be descriptive enough. test name should describe what exactly it tests, for example `should <result> when <precondition>`

 this is a common mistake across the project

dHihich replied:
also tests should be as granular as possible and should normally test one thing. For example, in order to avoid duplication you can render a component in `beforeEach` block and then just use the rendered test component in test scenarios

dHihich reviewed (changes requested):
Your UI doesn't match design provided in requirements. Please pay more attention to styling. It should look more or less like on mockups

Not completed yet. I'll have another look when you adjust styling

RamanReut committed:
Rework styles

RamanReut committed:
Rework unit tests; Add eslinet eol-last rule

dHihich commented on .gitignore:
```diff
@@ -21,3 +21,4 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+yarn.lock
```
why `yarn.lock` ignored?

dHihich commented on package.json:
```diff
@@ -3,17 +3,23 @@
   "version": "0.1.0",
```
![image](https://user-images.githubusercontent.com/30228677/91706512-2a3d6380-eb87-11ea-9df6-ea1d3bcf4ecc.png)

![image](https://user-images.githubusercontent.com/30228677/91706547-33c6cb80-eb87-11ea-9edd-32aa0914484d.png)

errors in console

dHihich replied:
![image](https://user-images.githubusercontent.com/30228677/91706763-83a59280-eb87-11ea-9199-8fdc70a4b8d9.png)

dHihich commented on src/features/layout/Header.tsx:
```diff
@@ -88,12 +102,20 @@ export default function Header({
                     >
                         <Grid item>
                             <TextField
```
pls extract into separate component

dHihich commented on src/features/layout/Header.tsx:
```diff
@@ -62,10 +74,12 @@ export default function Header({
                         <Grid item>
-                            <MenuButton 
-                                state={navbarPosition}
+                            <IconButton
```
I think it should be shown for mobile only, right? on middle and large size screens you show chevron icon on the bottom which controls menu visibility

dHihich commented on src/features/layout/NavbarList.tsx:
```diff
@@ -8,48 +8,124 @@ import SettingsIcon from '@material-ui/icons/Settings'
+                </Grid>
+            </Grid>
+            <Box className={classes.backgroundIconWrapper}>
+                <Icon className={classes.backgroundIcon} color='primary'>
```
extract icon into separate file

dHihich commented on src/features/layout/Path.tsx:
```diff
@@ -20,6 +26,7 @@ export default function Path() {
             maxItems={4}
+            separator={<SeparatorIcon fontSize='small'></SeparatorIcon>}
         >
             {createPathPoints(['main', 'data'])}
```
hardcoded value

dHihich replied:
link is non-clickable (no route specified)

dHihich replied:
please support a case when route and label might be different. for example `/data/my-custom-data-route` should be displayed as `Data -> Custom`

dHihich replied:
you can fix Breadcrumbs in a separate branch (and with separate PR) later on 'cause it looks like a separate task

dHihich reviewed (commented):
looks good despite of a few small comments

RamanReut committed:
Add expand animation to search input on hover;
Extract search input into separate component;
Hide menu button for large and middle monitor
